### PR TITLE
Make forms and AJAX code csrf safe

### PR DESCRIPTION
**Pull request by sravfeyn**
This makes forms and AJAX code csrf safe. This is #9227 minus actual middleware part. This will make it easy to manage #9227 on staging without too many conflicts. 
cc @esoergel FYI @snopoke @czue

### CODE REVIEW

File: corehq/apps/reports/static/reports/ko/saved_reports.js [hunk 1]
```diff
@@ -148,6 +148,9 @@ var ReportConfigsViewModel = function (options) {
         $.ajax({
             type: "DELETE",
             url: options.saveUrl + '/' + config._id(),
+            beforeSend: function(xhr, settings) {
```
Comment (proteusvacuum) [inline corehq/apps/reports/static/reports/ko/saved_reports.js]: seems like we'd have to do this for every `$.ajax` call (including `$.post`, etc), which seems a little annoying / non-DRY. Is there a way we can add this to the global `jQuery` post config?

Comment (sravfeyn) [inline corehq/apps/reports/static/reports/ko/saved_reports.js]: There's `ajaxSetup` that I could have used, but thought this approach was better. I will change this to use `ajaxSetup`.

File: corehq/apps/sms/templates/sms/languages.html [hunk 2]
```diff
@@ -20,6 +20,7 @@
 {% block js %}{{ block.super }}
     <script src="{% static 'hqwebapp/js/lib/jquery-ui/jquery-ui-1.8.16.min.js' %}"></script>
     <script src="{% static 'style/ko/knockout_bindings.ko.js' %}"></script>
+    <script src="{% static 'hqwebapp/js/ajax_csrf_setup.js' %}"></script>
```
Comment (proteusvacuum) [inline corehq/apps/sms/templates/sms/languages.html]: This seems like a hard problem where this file is now a dependency for `main.js`... after talking to @sravfeyn sounds like this is a tradeoff between one form of DRYness with another.

File: corehq/apps/hqwebapp/static/hqwebapp/js/main.js [hunk 3]
```diff
@@ -158,7 +161,7 @@ var SaveButton = {
                         type: 'POST',
                         dataType: 'json',
                         data: $form.serialize(),
-                        success: options.success
+                        success: options.success,
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/static/hqwebapp/js/main.js]: nit, but trailing commas in JS fail on some browsers.

**Comment by proteusvacuum** [comment]
:+1: nothing blocking. 
:lock: :cookie: :cookie: :cookie: :lock:

**Comment by sravfeyn** [comment]
@proteusvacuum responded.

**Comment by proteusvacuum** [comment]
Looks like you can maybe move the `ajax_csrf_setup.py` to a base template now, since you don't call that method anywhere. Will let you do that in a follow up PR if you think it's worth it.

**Pull request merged**